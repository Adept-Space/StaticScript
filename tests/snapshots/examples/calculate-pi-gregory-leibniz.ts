
{
    // Use the Gregory-Leibniz series.
    function calculatePI(cycles: number): number {
        let pi = 0;
        let i = 1;

        while (i < cycles) {
            pi = pi + (4 / i);
            pi = pi - (4 / (i + 2));

            i = i + 4;
        }

        return pi;
    }

    console_log("Gregory-Leibniz series");
    console_log(calculatePI(500000));
}
