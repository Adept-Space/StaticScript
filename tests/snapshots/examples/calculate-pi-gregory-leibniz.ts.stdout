Gregory-Leibniz series
3.141589
